contracts/Warehouse.ligo

```diff
@@ -84,13 +84,13 @@ function assign (const params: assign_parameter; var storage: storage): return i
                     failwith ("NO_AVAILABLE_ITEM");
                 } else {
                     const inventory : contract (inventory_parameter) =
-                        case (Tezos.get_entrypoint_opt ("%assign_item", params.0) : option (contract (inventory_parameter))) of
+                        case (Tezos.get_entrypoint_opt("%assign_item", params.0) : option (contract (inventory_parameter))) of
                             Some (contract) -> contract
                             | None -> (failwith ("CONTRACT_NOT_FOUND") : contract (inventory_parameter))
                         end;
 
                     const action : inventory_parameter = Assign_item (record [
-                        data = (Map.empty: map (string, string));
+                        data = fi.data;
```

Comment:
when we assign an item to a user we don't actually take the data field from the warehouse and assign it to the data field in the inventory. We set this value to an empty map. This change actually introduces a regression, can you revert it?

If we create 1 warehouse item with a total quantity of 1 million, and we duplicate the data field from warehouse into every 1 million inventory item, if one item costs 0.05tz to store, this change would actually cost us 50 000tz, or 100 000 euros at today's rate :)

The inventory data field should only store the changes that have changed from the warehouse. To get the final state state of an inventory item we need to:
1. take the warehouse item data field
2. merge in the state from the inventory item data field

so the previous code was actually correct. Can you revert?